scripts/util/pancancer_util.R

```diff
@@ -153,3 +153,40 @@ add_arrow_label <- function(p, x, y, label, offset = c(0, 0, 0, 0)) {
              size = 0.18, arrow = arrow(length = unit(0.05, "cm")))
   return(p)
 }
+
+my.mean <- function(x, indices) {
+  # Function to use as `statistic` argument in call to boot()
+  # This will enable a bootstrap estimate of the mean and confidence intervals
+  #
+  # The Function is taken from "Chapter 3 R Bootstrap Examples" by Bret Larget
+  # http://www.stat.wisc.edu/~larget/stat302/chap3.pdf
+  return( mean( x[indices] ) )
+}
+
+get_boot <- function(weight, low = TRUE, R = 1000, min_samples = 5) {
```

Comment:
In general I use [`Hmisc::smean.cl.boot`](https://www.rdocumentation.org/packages/Hmisc/versions/4.0-3/topics/smean.sd), which is what ggplot2 uses. Would this function make it easier?

Reply:
Looking into this now - thanks for pointing me to the function. I do believe `Hmisc::smean.cl.boot()` is slightly cleaner than a combination of `boot()` and `boot.ci()`. It is also slightly faster and returns a similar result.

```R
> test <- mut_weight_df[mut_weight_df$Hugo_Symbol == 'KRAS', ]
> 
> Hmisc::smean.cl.boot(test$weight)
     Mean     Lower     Upper 
0.8142939 0.8053043 0.8242565 
> boot::boot.ci(
+   boot::boot(test$weight, statistic = my.mean, R = 1000),
+   type = "perc")
BOOTSTRAP CONFIDENCE INTERVAL CALCULATIONS
Based on 1000 bootstrap replicates

CALL : 
boot::boot.ci(boot.out = boot::boot(test$weight, statistic = my.mean, 
    R = 1000), type = "perc")

Intervals : 
Level     Percentile     
95%   ( 0.8043,  0.8247 )  
Calculations and Intervals on Original Scale
> 
> system.time(Hmisc::smean.cl.boot(test$weight))
   user  system elapsed 
  0.012   0.000   0.011 
> system.time(boot::boot.ci(
+   boot::boot(test$weight, statistic = my.mean, R = 1000),
+   type = "perc"))
   user  system elapsed 
  0.016   0.000   0.018
```

The next commit will switch to `Hmisc`